test/test_sync_protocol_codec.cpp

```diff
@@ -0,0 +1,224 @@
+/*************************************************************************
+ *
+ * Copyright 2016 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ **************************************************************************/
+
+#include "testsettings.hpp"
+#include "test.hpp"
+
+#include <realm/sync/noinst/protocol_codec.hpp>
+
+#include <string>
+#include <memory>
+
+using namespace realm;
+using namespace realm::test_util;
+using namespace realm::util;
+
+template <class A, class B>
+bool compare_out_string(const A& expected, const B& out, const std::shared_ptr<util::Logger>& logger)
+{
+    if (expected.size() == out.size() && memcmp(expected.data(), out.data(), out.size()) == 0) {
```

Comment:
if you wanted to make this a little more idiomatic, at least to my eyes you could do something like
```
template <class A, class B>
void check_out_string(const A& expected, const B& out, unit_test::TestContext& test_context)
{
    CHECK_EQUAL(std::string_view(std::data(expected), std::size(expected)), std::string_view(std::data(out), std::size(out)));
}
```

Then the CHECK_EQUAL macro would just work as expected.

Reply:
That works great - thanks for the recommendation